lib/services/role.go

```diff
@@ -1691,6 +1691,17 @@ func (set RoleSet) CanPortForward() bool {
 	return false
 }
 
+// RecordDesktopSession returns true if a role in the role set has enabled
+// desktop session recoring.
+func (set RoleSet) RecordDesktopSession() bool {
```

Comment:
Note to reviewers: this uses the same logic that we use throughout the RBAC system for  boolean options that default to true. That is, as long as one of the user's roles has enabled the feature, it will be enabled.

I *think* this is what we want here, but it does mean it may be difficult to disable session recording, as *all* of the user's roles would have to explicitly disable it.

Seem reasonable or might we need a more flexible approach here?

Reply:
Good question. I was just looking through the code and documentation and it looks like we don't truly have precedence for this feature with SSH session recording, since it's set at the cluster-level and so there aren't such role-level conflicts.

This raises the question -- do we truly want role-level configurability for session recording? I can see it being a useful feature, but barring a specific request from a customer I think we should mimic SSH session recording and configure it in `windows_desktop_service` to start.

Reply:
Yes, we truly want it. This was discussed and approved in RFD 33.

In a nutshell, the reason for having this associated with roles as opposed to cluster wide is that some users may have long (several hours) desktop sessions and not want to save the very large recording (not an issue with smaller SSH recordings).

Reply:
Will there be accompanying docs changes that highlight this difference in behavior? I could see users being slightly confused about this at first.

Reply:
@rosstimothy FTR we're currently discussing this in an email thread with Sasha and Xin. Slow going right now due to Sasha being OOO.